server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -505,6 +516,42 @@ private void handleBadRequest(String uri, RestRequest.Method method, RestChannel
         }
     }
 
+    /**
+     * Attempts to extracts auth token and login.
+     * 
+     * @returns false if there was an error and the request should not continue being dispatched
+     * */
+    private boolean handleAuthenticateUser(final RestRequest request, final RestChannel channel) {    
+        if (!FeatureFlags.isEnabled(FeatureFlags.IDENTITY)) {
+            return true;
+        }
+
+        try {
+            final AuthToken token = RestTokenExtractor.extractToken(request);
+            // If no token was found, continue executing the request
+            if (token == null) {
+                return true;
+            }
+            final Subject currentSubject = identityService.getSubject();
+            currentSubject.login(token);
+            logger.info("Logged in as user " + currentSubject);
+        } catch (final Exception e) {
+            try {
+                final BytesRestResponse bytesRestResponse = BytesRestResponse.createSimpleErrorResponse(
+                    channel,
+                    RestStatus.UNAUTHORIZED,
+                    e.getMessage()
+                );
+                channel.sendResponse(bytesRestResponse);
+            } catch (final Exception _ignored) {
+                // TODO: clean this up
+            }
+            return false;
+        }
+
+        return true;
```

Comment:
By default we should always false and return true only when the subject is authenticated (at L537)